File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> res(nums.begin(), nums.end());
        int longest = 0;
        for(auto x : res){
            if(res.find(x - 1) == res.end()){
                int len = 1;
                while(res.find(x + len) != res.end()){
                    len++;
                }
                longest = max(longest, len);
            }
        }
        return longest;
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)

        if n < 1:
            return 0
        dp = [1] * (n + 1)

        res = 1
        for i in range(n):
            for j in range(i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j] + 1)
                    res = max(res, dp[i])

        return res